src/main/java/eu/europeana/metis/sandbox/entity/HarvestingParameterEntity.java

```diff
@@ -0,0 +1,180 @@
+package eu.europeana.metis.sandbox.entity;
+
+import eu.europeana.metis.sandbox.common.HarvestProtocol;
+import eu.europeana.metis.sandbox.dto.FileHarvestingDto;
+import eu.europeana.metis.sandbox.dto.HttpHarvestingDto;
+import eu.europeana.metis.sandbox.dto.OAIPmhHarvestingDto;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.JoinColumn;
+import javax.persistence.Enumerated;
+import javax.persistence.EnumType;
+
+/**
+ * Entity to map harvesting_parameters table
+ */
+@Entity
+@Table(name = "harvesting_parameter")
+public class HarvestingParameterEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne(cascade = CascadeType.PERSIST)
+    @JoinColumn(name = "dataset_id", referencedColumnName = "datasetId")
+    private DatasetEntity datasetId;
+
+    @Enumerated(EnumType.STRING)
+    private HarvestProtocol harvestProtocol;
+
+    private String fileName;
+
+    private String fileType;
+
+    private String url;
+
+    private String setSpec;
+
+    private String metadataFormat;
+
+    /**
```

Comment:
remove this constructor so only the interface is used.
```
  /**
     * Parameterized constructor
     * @param datasetId The id of the dataset associated to these harvesting parameters
     * @param harvestProtocol The type of harvesting done
     * @param fileName The name of ile that was harvested (if it exists)
     * @param fileType The type of the file that was harvested (if it exists)
     * @param url The url used to be harvested (if it exists)
     * @param setSpec The setspec used for the harvesting (if it exists)
     * @param metadataFormat The metadata format used for the harvesting (if it exists)
     */
    public HarvestingParameterEntity(DatasetEntity datasetId, HarvestProtocol harvestProtocol, String fileName, String fileType,
                                     String url, String setSpec, String metadataFormat) {
        this.datasetId = datasetId;
        this.harvestProtocol = harvestProtocol;
        this.fileName = fileName;
        this.fileType = fileType;
        this.url = url;
        this.setSpec = setSpec;
        this.metadataFormat = metadataFormat;
    }
```

Reply:
Done